scipy/optimize/optimize.py

```diff
@@ -2102,7 +2105,11 @@ def _minimize_scalar_golden(func, brack=None, args=(),
     f1 = func(*((x1,) + args))
     f2 = func(*((x2,) + args))
     funcalls += 2
-    while (numpy.abs(x3 - x0) > tol * (numpy.abs(x1) + numpy.abs(x2))):
+    nit = 0
+    for i in xrange(maxiter):
+        if ((numpy.abs(x3 - x0) <= tol * (numpy.abs(x1) + numpy.abs(x2))) or
+                (x1 == x2)):
```

Comment:
`x1 == x2` does not necessarily imply successful convergence.
It's also not clear that it implies an endless loop, as it seems the interval will still continue shrinking --- without an explicit proof that it does, it would be better to simply not check for it and rely on maxiter for termination.